lumicks/pylake/tests/test_imaging_confocal/test_kymo.py

```diff
@@ -6,303 +6,300 @@
+        empty_kymograph.save_tiff("test")
+
+    with pytest.raises(RuntimeError):
+        empty_kymograph.plot_rgb()
+
+    assert empty_kymograph.red_image.shape == (5, 0)
+    with pytest.deprecated_call():
+        empty_kymograph.json
+    with pytest.deprecated_call():
+        assert empty_kymograph.has_fluorescence
+    with pytest.deprecated_call():
+        assert not empty_kymograph.has_force
+    assert empty_kymograph.infowave.data.size == 0
+    assert empty_kymograph.pixels_per_line == 5
+    assert empty_kymograph.red_image.size == 0
+    assert empty_kymograph.rgb_image.size == 0
+
+
+def test_damaged_kymo(test_kymos):
+    # Assume the user incorrectly exported only a partial Kymo
+    kymo = test_kymos["damaged"]
+    kymo_reference = np.transpose([[2, 0, 0, 0, 2], [0, 0, 0, 0, 0], [1, 0, 0, 0, 1], [0, 1, 1, 1, 0]])
+
+    with pytest.warns(RuntimeWarning):
+        assert kymo.red_image.shape == (5, 3)
+    np.testing.assert_allclose(kymo.red_image.data, kymo_reference[:, 1:])
 
 
 @cleanup
-def test_plotting(h5_file):
-    f = pylake.File.from_h5py(h5_file)
-    if f.format_version == 2:
-        kymo = f.kymos["Kymo1"]
-
-        kymo.plot_red()
-        # The following assertion fails because of unequal line times in the test data. These
-        # unequal line times are not typical for BL data. Kymo nowadays assumes equal line times
-        # which is why the old version of this test fails.
-        # np.testing.assert_allclose(np.sort(plt.xlim()), [-0.5, 3.5], atol=0.05)
-        np.testing.assert_allclose(plt.xlim(), [-0.515625, 3.609375])
-        np.testing.assert_allclose(np.sort(plt.ylim()), [0, 0.05])
-
-        # test original kymo is labeled with microns and
-        # that kymo calibrated with base pairs has appropriate label
-        assert plt.gca().get_ylabel() == r"position ($\mu$m)"
-        kymo_bp = kymo.calibrate_to_kbp(10.000)
-        kymo_bp.plot_red()
-        assert plt.gca().get_ylabel() == "position (kbp)"
+def test_plotting(test_kymos):
+    kymo = test_kymos["Kymo1"]
+
+    plt.figure()
+    kymo.plot_red()
+    # # The following assertion fails because of unequal line times in the test data. These
+    # # unequal line times are not typical for BL data. Kymo nowadays assumes equal line times
+    # # which is why the old version of this test fails.
+    # np.testing.assert_allclose(np.sort(plt.xlim()), [-0.5, 3.5], atol=0.05)
+
+    # # The following two assertions are fragile (they seem to break when the test file is moved
+    # # to a different directory). These have been replaced with tests asserting on the actual
+    # # image data.
+    # np.testing.assert_allclose(plt.xlim(), [-0.515625, 3.609375])
+    # np.testing.assert_allclose(np.sort(plt.ylim()), [0, 0.05])
+    image = plt.gca().get_images()[0]
+    np.testing.assert_allclose(image.get_array(), kymo.red_image)
+    # np.testing.assert_allclose(image.get_extent(), [0, 0.197 * 3, 0.191 * 4, 0])
```

Comment:
While I agree that maybe testing the limits is not a good idea, we do explicitly set extents. So may be good to indeed check these.

Reply:
yes I agree. I had written that line that's commented out (oops) incorrectly. It's fixed now